src/main/java/Parser.java

```diff
@@ -0,0 +1,122 @@
+import java.time.LocalDate;
+
+/**
+ * Handles parsing of user inputs.
+ */
+public class Parser {
+
+    /**
+     * Stores the parsed user inputs and prints response messages for user.
+     * @param userInput User input read by scanner.
+     * @param duke Duke object.
+     * @return Duke response message.
+     */
+    public String parse(String userInput, Duke duke) {
```

Comment:
I think you can consider shortening this method (as it seems quite long) by creating other private methods as helper functions! 👍

Reply:
I agree that it's quite long hahaha I'll work on it! thanks :))